implants/lib/transport/src/mock.rs

```diff
@@ -0,0 +1,41 @@
+use anyhow::Result;
+use pb::c2::*;
+use std::sync::mpsc::{Receiver, Sender};
+
+use mockall::{mock, predicate::*};
+
+mock! {
```

Comment:
This seems like magic how does this work? Can we add some comments about what's going on?

Reply:
Honestly we're not even using it here, but we can add comments where we will actually use it (integration tests for eldritch & imix tests)

Reply:
That'd be awesome.
Maybe a comment here to ref those examples in case someone ends up her for some reason.

Reply:
We don't have any of these tests yet, but will do buckaroo